src/main/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctions.scala

```diff
@@ -165,6 +165,22 @@ object PipelineFunctions {
       .alias(defaultAlias)
   }
 
+  /**
+   * Builds the Map to be used in modifyStruct changeInventory to clean a new_cluster schema
+   * @param df df that contains the prefixed field to cleanse
+   * @param newClusterPrefix path to new_cluster including the "new_cluster" such as "new_settings.new_cluster"
+   * @return Map[String, Column] for changeInventory
+   */
+  def newClusterCleaner(df: DataFrame, newClusterPrefix: String): Map[String, Column] = {
+    Map(
+      s"${newClusterPrefix}.custom_tags" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.custom_tags"),
+      s"${newClusterPrefix}.spark_conf" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.spark_conf"),
+      s"${newClusterPrefix}.spark_env_vars" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.spark_env_vars"),
+      s"${newClusterPrefix}.aws_attributes" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.aws_attributes"),
+      s"${newClusterPrefix}.azure_attributes" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.azure_attributes")
```

Comment:
should we add `gcp_attributes` in advance? Maybe just add comment not to forget it

Reply:
yeah that'll be broad to do when we add support for GCP -- we'll have to do that in several places.